tall_brain/Assignment_7/calculator.py

```diff
@@ -29,9 +29,55 @@ def calculate(self, prob):
         elif o == "/":
             return self.div(c1, c2)
 
+    def AdvCalculate(self, prob):
+        if "+" in prob:
+            adv_parsed_prob = prob.split("+")
+            c1 = adv_parsed_prob[0]
+            c_1 = c1.strip()
+            if c_1.isnumeric():
+                c_1 = int(c_1)
+            c2 = adv_parsed_prob[1]
+            c_2 = c2.strip()
+            if c_2.isnumeric():
+                c_2 = int(c_2)
+            return self.add(c_1, c_2)
+        elif "-" in prob:
+            adv_parsed_prob = prob.split("-")
+            c1 = adv_parsed_prob[0]
+            c_1 = c1.strip()
+            if c_1.isnumeric():
+                c_1 = int(c_1)
+            c2 = adv_parsed_prob[1]
+            c_2 = c2.strip()
+            if c_2.isnumeric():
+                c_2 = int(c_2)
```

Comment:
All of this code is duplicated in each of the if/elif blocks. This can be refactored to not duplicate logic.